Add new folder for the x-Days-of-Code-Counter

NazaNEYn opened the pull request:
A small script to track your progress for the "100 Days of Code" challenge. <br>
But the number of the days isn't a fixed number. you will be asked a question and then you can choose your number and the code will be flexible.<br>
It has a progress bar.<br>
It also has a short readme file. 


https://github.com/user-attachments/assets/a1edc751-a59c-490d-8ea9-0c3eeac57536

NazaNEYn committed:
Add new folder for the x-Days-of-Code-Counter

NazaNEYn committed:
Added the unit-converter file

NazaNEYn commented:
Also added new folder, the unit-converter folder where you can convert 3 units. <br>
1. fahrenheit to celsius
2. miles to kilometers
3. inches to centimeter

mfranzon commented on scripts/unit-converter/length_conversions.py:
```diff
@@ -0,0 +1,18 @@
+#  and in_to_cm()
```
remove comment, probably a leftover

mfranzon commented on scripts/unit-converter/main.py:
```diff
@@ -0,0 +1,61 @@
+from temperature_conversions import fahrenheit_to_celsius
+from length_conversions import miles_to_km, inches_to_centimeter
+
+# main.py
```
remove comment

mfranzon commented on scripts/unit-converter/length_conversions.py:
```diff
@@ -0,0 +1,18 @@
+
+def inches_to_centimeter(inch):
+    cm = 2.54
+    return inch * cm
```
In general, single scope functions should be put into a common module like `utils.py`. I suggest to add it into `unit-converter/` and put into `miles_to_km`, `inches_to_centimeter` and `fahrenheit_to_celsius`

mfranzon commented on scripts/unit-converter/temperature_conversions.py:
```diff
@@ -0,0 +1,5 @@
+
+
+def fahrenheit_to_celsius(far):
+    return (far - 32) * 5 / 9
```
See comment above

mfranzon commented on scripts/x-Days-of-Code-Counter/count_down.py:
```diff
@@ -0,0 +1,13 @@
+class CounterDown:
```
Consider to have a single module `Counter.py` in which defining `CounterUp` and `CounterDown`

mfranzon commented on scripts/x-Days-of-Code-Counter/counter.py:
```diff
@@ -0,0 +1,18 @@
+class CounterUp:
```
See comment above

mfranzon commented on scripts/x-Days-of-Code-Counter/main.py:
```diff
@@ -0,0 +1,34 @@
+counter_down = CounterDown(user_max_day)
+
+is_counting = True
+while is_counting:
```
In general I like this idea, if you agree I have an idea to replace the bar animation with a hourglass that drops some grains proportional to the number of days

NazaNEYn committed:
fix: removed leftover commentes and the unnecessary files and moved them to utils.py

NazaNEYn committed:
Update main.py with new imports from utils.py

NazaNEYn committed:
Fix: created a single module Counter.py and replaced the progressbar with the hourglass

NazaNEYn commented:
I removed the leftover comments, fixed the modules and replaced the progress bar with the hourglass.<br>
Generating some cool hourglass wasn't easy. So let's see if you like it :D

mfranzon committed:
adding docstrings and module structure

mfranzon committed:
rethink hourglass animation about days of code

mfranzon commented:
Hi @NazaNEYn , I push a new version, I think it is funny having an animated version instead of hitting enter. Let me know what do you think

NazaNEYn committed:
Fixed the code according to the pr review

NazaNEYn committed:
Address review comments and docstring additions

NazaNEYn committed:
added the text cleaner script

NazaNEYn commented:
@mfranzon added the text cleaner script. it's very basic. I just lowercase the words, remove punctuations and extra white spaces and save the file in a new file.
I only used the `string` module instead of `re` module since that isn't beginner friendly. 
So it's a very basic text cleaner :D

mfranzon commented on scripts/Text-Cleaner/main.py:
```diff
@@ -0,0 +1,38 @@
+
+    # breaking the string into a list of words.
+    list_of_words = content.split()
+
```
After split the sentence in a list of words you could search and remove something, like a special characters for example. What do you think?

NazaNEYn replied:
what do you mean? I don't get it :D
